guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/CacheSerializer.java

```diff
@@ -46,10 +105,26 @@ public void serializeWithType(Cache<?, ?> value, JsonGenerator gen, SerializerPr
         _writeContents(value, gen, ctxt);
         typeSer.writeTypeSuffix(gen, typeIdDef);
     }
+    
+    /*
+    /**********************************************************
+    /* Internal helper methods
+    /**********************************************************
+     */
 
-    // Just a stub in case we have time to implement proper (if optional) serialization
-    protected void _writeContents(Cache<?, ?> value, JsonGenerator g, SerializerProvider ctxt)
-        throws JacksonException
+    protected void _writeContents(Cache<?, ?> cache, JsonGenerator gen, SerializerProvider provider)
+        throws IOException
     {
-    }    
+        final Set<String> ignored = _ignoredEntries;
+        for (Map.Entry<?, ?> entry : cache.asMap().entrySet()) {
+            Object key = entry.getKey();
+            Object value = entry.getValue();
+            // ignore? 
+            if (ignored != null && ignored.contains(key.toString())) {
+                continue;
+            }
+            gen.writeFieldName(key.toString());
```

Comment:
This is different from regular maps, which supports a set of JDK types, defined in `StdKeySerializers`.
You may want to have a look at `MapSerializer` which fetches `_keySerializer` (and one for values).